### PR TITLE
add pstrain

### PR DESCRIPTION
**Pull request by wshuai294**
Describe your pull request here
Add a strain-profiling method: PStrain.
----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

### General instructions

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

### Instructions for avoiding API, ABI, and CLI breakage issues
Conda is able to record and lock (a.k.a. pin) dependency versions used at build time of other recipes.
This way, one can avoid that expectations of a downstream recipe with regards to API, ABI, or CLI are violated by later changes in the recipe.
If not already present in the meta.yaml, make sure to specify `run_exports` (see [here](https://bioconda.github.io/contributor/linting.html#missing-run-exports) for the rationale and comprehensive explanation).
Add a `run_exports` section like this:

```yaml
build:
  run_exports:
    - ...

```

with `...` being one of:

| Case                             | run_exports statement                                               |
| -------------------------------- | ------------------------------------------------------------------- |
| semantic versioning              | `{{ pin_subpackage("myrecipe", max_pin="x") }}`     |
| semantic versioning (0.x.x)      | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}`   |
| known breakage in minor versions | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| known breakage in patch versions | `{{ pin_subpackage("myrecipe", max_pin="x.x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| calendar versioning              | `{{ pin_subpackage("myrecipe", max_pin=None) }}`    |

while replacing `"myrecipe"` with either `name` if a `name|lower` variable is defined in your recipe or with the lowercase name of the package in quotes.

### Bot commands for PR management

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

Note that the <code>@BiocondaBot please merge</code> command is now depreciated. Please just squash and merge instead.

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by wshuai294** [comment]
@BiocondaBot please add label

**Comment by wshuai294** [comment]
@bioconda/core

File: recipes/pstrain/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  license: MIT
+  summary: An Iterative Microbial Strains Profiling Algorithm for Shotgun Metagenomic Sequencing Data
+
+test:
+  commands:
+    - pstrain --help
+
+extra:
+  skip-lints:
+    - should_be_noarch_generic
```
Comment (bgruening) [inline recipes/pstrain/meta.yaml:43]: any ideas why we can not have a `noarch: generic`.

Comment (wshuai294) [reply recipes/pstrain/meta.yaml:44]: Hello, I have tested `noarch: generic` and `noarch: python`, but both of them failed. I think the reason might be that dependent packages like Metaphlan3 are not platform-independent.

Comment (wshuai294) [reply recipes/pstrain/meta.yaml:44]: Hello, I have tested `noarch: generic` and it did not work. I wonder if it is necessary to add this statement. Thanks!

Comment (martin-g) [reply recipes/pstrain/meta.yaml:44]: How did it break ?

File: recipes/pstrain/build.sh [hunk 2]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+
+mkdir -p $PREFIX/bin
+
+cp $SRC_DIR/scripts/*py $PREFIX/bin
```
Comment (martin-g) [inline recipes/pstrain/build.sh:5]:
```suggestion
cp $SRC_DIR/scripts/*.py $PREFIX/bin
```

File: recipes/pstrain/build.sh [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+cp $SRC_DIR/scripts/*sh $PREFIX/bin
```
Comment (martin-g) [inline recipes/pstrain/build.sh:6]:
```suggestion
cp $SRC_DIR/scripts/*.sh $PREFIX/bin
```

File: recipes/pstrain/build.sh [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+cp $SRC_DIR/scripts/*jar $PREFIX/bin
```
Comment (martin-g) [inline recipes/pstrain/build.sh:7]:
```suggestion
cp $SRC_DIR/scripts/*.jar $PREFIX/bin
```

File: recipes/pstrain/meta.yaml (hunk 1 above)
Comment (martin-g) [reply recipes/pstrain/meta.yaml]: I'd not expect to see such lint error with `noarch: generic`

Comment (martin-g) [reply recipes/pstrain/meta.yaml]: https://bioconda.github.io/contributor/linting.html#should-be-noarch-generic
I think your package contains both Java (*.jar) and Python code (*.py). I guess this is the reason why it fails ...

Comment (wshuai294) [reply recipes/pstrain/meta.yaml:44]: That makes sense. So, could I just remove `noarch: generic` and skip the lint check?

Comment (martin-g) [reply recipes/pstrain/meta.yaml:44]: I don't have much experience with this. Let's see what more experienced devs will say!

Comment (wshuai294) [reply recipes/pstrain/meta.yaml:44]: Okay. Thanks!

**Review by martin-g — APPROVED** [review]

**Pull request merged**